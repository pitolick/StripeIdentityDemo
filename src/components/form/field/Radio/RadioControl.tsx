/**
 * ラジオボタン制御コンポーネント（react-hook-form）
 * @module RadioControl
 * @todo カードスタイルのときにTabでフォーカスした場合にアウトラインがでないのを修正
 */
import React, { forwardRef } from 'react'
import { Radio, RadioProps } from './Radio'
import { InputBox, InputBoxProps } from '@/components/form/parts'

/**
 * ラジオボタンコンポーネントのプロパティ
 * @typedef {Object} RadioControlProps
 * @property {React.InputHTMLAttributes<HTMLInputElement>} - ラジオボタンタグのHTML属性
 */

export interface RadioControlProps extends React.InputHTMLAttributes<HTMLInputElement>, Omit<InputBoxProps, 'children'> {
  /**
   * リスト
   */
  lists: RadioProps[]
  /**
   * ラベルのタイプ
   */
  labelType?: 'default'
  /**
   * 並び方
   */
  direction?: 'row' | 'col'
  /**
   * ラジオボタンの位置
   */
  radioPosition?: 'center' | 'top' | 'hidden'
  /**
   * ラジオボタンの色
   */
  radioColor?: 'black' | 'red' | 'blue' | 'gray'
}

/**
 * ラジオボタン制御コンポーネント（react-hook-form）
 * @param {RadioControlProps} props - ラジオボタンコンポーネントのプロパティ
 * @returns {JSX.Element} ラジオボタンコンポーネント
 */
export const RadioControl = forwardRef<HTMLInputElement, RadioControlProps>((props, ref) => {
  const { lists, nameId, label, note, required = undefined, error, labelType = 'default', direction = 'row', radioPosition = 'center', radioColor = 'blue', contentWidth, ...radioControlProps } = props

  const radioList = lists.map((list, index) => {
    const { labelContents, radioClassName, labelClassName, ...radioProps } = list
    // class名
    let radioClassNameProps = `${radioClassName ? radioClassName : ''}`
    let labelClassNameProps = `${labelClassName ? labelClassName : ''} inline-block relative pl-5`

    // ラジオボタンの位置
    switch (radioPosition) {
      case 'center':
        radioClassNameProps = radioClassNameProps + ' absolute margin-auto top-1/2 transform -translate-y-1/2'
        break
      case 'top':
        radioClassNameProps = radioClassNameProps + ' absolute margin-auto'
        // ラベルのタイプによって分岐
        switch (labelType) {
          case 'default':
            radioClassNameProps = radioClassNameProps + ' top-1'
            break
        }
        break
      case 'hidden':
        radioClassNameProps = 'hidden'
        break
    }

    // エラー時のclass名
    let errorLabelClass = ''
    let errorRadioClass = ''

    // ラベルのタイプ
    switch (labelType) {
      case 'default':
        radioClassNameProps = radioClassNameProps + ' left-0'
        errorLabelClass = error ? ' text-red' : ''
        errorRadioClass = error ? ' bg-red-pale text-red border-red' : ''
        break
    }

    // ラジオボタンの色
    switch (radioColor) {
      case 'black':
        radioClassNameProps = radioClassNameProps + ' checked:text-black focus:text-black'
        break
      case 'red':
        radioClassNameProps = radioClassNameProps + ' checked:text-red focus:text-red'
        break
      case 'blue':
        radioClassNameProps = radioClassNameProps + ' checked:text-blue focus:text-blue'
        break
      case 'gray':
        radioClassNameProps = radioClassNameProps + ' checked:text-gray-700 focus:text-gray-700'
        break
    }

    // カードコンポーネント
    const Wrapper: React.FC<{ children: React.ReactNode }> = ({ children }) => {
      return <>{children}</>
    }

    return (
      <div key={`${nameId}_${index}`} className={`${direction === 'col' && 'w-full'} ${direction === 'row' && contentWidth === 'full' && 'flex-auto'}`}>
        <Wrapper>
          <Radio
            ref={ref}
            labelContents={labelContents}
            radioClassName={`${radioClassNameProps} ${errorRadioClass}`}
            labelClassName={`${labelClassNameProps} ${errorLabelClass}`}
            wrapper={true}
            {...radioProps}
            {...radioControlProps}
          />
        </Wrapper>
      </div>
    )
  })

  return (
    <InputBox nameId={nameId} label={label} note={note} required={required} error={error} contentWidth={contentWidth}>
      {/* ラジオボタン */}
      <div className={`${'flex-' + direction} ${contentWidth === 'full' && 'w-full'} flex flex-wrap gap-3`}>{radioList}</div>
    </InputBox>
  )
})

RadioControl.displayName = 'RadioControl'
