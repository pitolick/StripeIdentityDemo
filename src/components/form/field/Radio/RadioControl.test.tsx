import React from 'react'
import { render } from '@testing-library/react'
import { RadioControl, RadioControlProps } from './RadioControl'

describe('RadioControl', () => {
  const props = {
    lists: [
      {
        value: 'option1',
        label: 'オプション1',
      },
      {
        value: 'option2',
        label: 'オプション2',
      },
    ],
  }

  it('正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" />)
    expect(container).toMatchSnapshot()
  })

  it('labelTypeプロパティがdefaultの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" labelType="default" />)
    expect(container).toMatchSnapshot()
  })

  it('labelTypeプロパティがdefaultでerrorがtrueの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" labelType="default" error={{ type: 'test' }} />)
    expect(container).toMatchSnapshot()
  })

  it('labelTypeプロパティがcardの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" />)
    expect(container).toMatchSnapshot()
  })

  it('labelTypeプロパティがcardでerrorがtrueの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" error={{ type: 'test' }} />)
    expect(container).toMatchSnapshot()
  })

  it('directionプロパティがrowの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" direction="row" />)
    expect(container).toMatchSnapshot()
  })

  it('directionプロパティがcolの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" direction="col" />)
    expect(container).toMatchSnapshot()
  })

  it('radioPositionプロパティがcenterの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" radioPosition="center" />)
    expect(container).toMatchSnapshot()
  })

  it('radioPositionプロパティがtopでlabelTypeがdefaultの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" radioPosition="top" labelType="default" />)
    expect(container).toMatchSnapshot()
  })

  it('radioPositionプロパティがtopでlabelTypeがcardの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" radioPosition="top" />)
    expect(container).toMatchSnapshot()
  })

  it('radioPositionプロパティがhiddenの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" radioPosition="hidden" />)
    expect(container).toMatchSnapshot()
  })

  it('radioColorプロパティがblackの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" radioColor="black" />)
    expect(container).toMatchSnapshot()
  })

  it('radioColorプロパティがredの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" radioColor="red" />)
    expect(container).toMatchSnapshot()
  })

  it('radioColorプロパティがblueの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" radioColor="blue" />)
    expect(container).toMatchSnapshot()
  })

  it('radioColorプロパティがgrayの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" radioColor="gray" />)
    expect(container).toMatchSnapshot()
  })

  it('contentWidthプロパティがfullの場合、正しくレンダリングされること', () => {
    const { container } = render(<RadioControl {...props} nameId="test" contentWidth="full" />)
    expect(container).toMatchSnapshot()
  })

  it('listsにradioClassName labelClassNameプロパティが指定されている場合、正しくレンダリングされること', () => {
    const { container } = render(
      <RadioControl
        {...props}
        nameId="test"
        lists={[
          {
            value: 'option1',
            label: 'オプション1',
            radioClassName: 'test-radio',
            labelClassName: 'test-label',
          },
        ]}
      />,
    )
    expect(container).toMatchSnapshot()
  })
})
