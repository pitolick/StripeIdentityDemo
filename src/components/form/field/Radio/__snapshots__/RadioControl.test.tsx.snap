// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadioControl contentWidthプロパティがfullの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="w-full flex-auto space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row w-full flex flex-wrap gap-3"
        >
          <div
            class="false flex-auto"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false flex-auto"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl directionプロパティがcolの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-col false flex flex-wrap gap-3"
        >
          <div
            class="w-full false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="w-full false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl directionプロパティがrowの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl labelTypeプロパティがcardでerrorがtrueの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  text-red cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue  bg-red-pale text-red border-red"
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  text-red cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue  bg-red-pale text-red border-red"
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
      <p
        class="text-sm text-red"
      />
    </div>
  </div>
</div>
`;

exports[`RadioControl labelTypeプロパティがcardの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl labelTypeプロパティがdefaultでerrorがtrueの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  text-red cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue  bg-red-pale text-red border-red"
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  text-red cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue  bg-red-pale text-red border-red"
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
      <p
        class="text-sm text-red"
      />
    </div>
  </div>
</div>
`;

exports[`RadioControl labelTypeプロパティがdefaultの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl listsにradioClassName labelClassNameプロパティが指定されている場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class="test-label inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent test-radio absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl radioColorプロパティがblackの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-black focus:text-black "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-black focus:text-black "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl radioColorプロパティがblueの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl radioColorプロパティがgrayの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-gray-700 focus:text-gray-700 "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-gray-700 focus:text-gray-700 "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl radioColorプロパティがredの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-red focus:text-red "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-red focus:text-red "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl radioPositionプロパティがcenterの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl radioPositionプロパティがhiddenの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent hidden left-0 checked:text-blue focus:text-blue "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent hidden left-0 checked:text-blue focus:text-blue "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl radioPositionプロパティがtopでlabelTypeがcardの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1 left-0 checked:text-blue focus:text-blue "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1 left-0 checked:text-blue focus:text-blue "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl radioPositionプロパティがtopでlabelTypeがdefaultの場合、正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1 left-0 checked:text-blue focus:text-blue "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1 left-0 checked:text-blue focus:text-blue "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`RadioControl 正しくレンダリングされること 1`] = `
<div>
  <div
    class="false space-y-2 false"
  >
    <div
      class="space-y-2 false"
    >
      <div
        class="flex w-full items-center justify-start gap-x-3"
      >
        <div
          class="flex-row false flex flex-wrap gap-3"
        >
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション1"
                type="radio"
                value="option1"
              />
            </label>
          </div>
          <div
            class="false false"
          >
            <label
              class=" inline-block relative pl-5  cursor-pointer"
            >
              <input
                class="focus:ring-transparent  absolute margin-auto top-1/2 transform -translate-y-1/2 left-0 checked:text-blue focus:text-blue "
                label="オプション2"
                type="radio"
                value="option2"
              />
            </label>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
